src/server/system_services/tier_server.js

```diff
@@ -101,11 +100,13 @@ function read_tier(req) {
         });
     });
 
-    return P.join(
+    return Promise.all([
             nodes_client.instance().aggregate_nodes_by_pool(_.compact(pool_names), req.system._id),
             nodes_client.instance().aggregate_data_free_by_tier([String(tier._id)], req.system._id)
-        )
-        .spread(function(nodes_aggregate_pool, available_to_upload) {
+        ])
+        .then(function(res) {
+            let nodes_aggregate_pool = res[0];
+            let available_to_upload = res[1];
```

Comment:
```suggestion
        .then(([nodes_aggregate_pool, available_to_upload]) {
```

Reply:
when doing that we get a lint error:
`Unexpected block statement surrounding arrow body; move the returned value immediately after the =>.eslint(arrow-body-style)`

Reply:
I was just missing the arrow `=>` Fixed the suggestion

Reply:
@guymguym not it, I manually added the arrow and got this lint error
so the code is:
```
        .then(([nodes_aggregate_pool, available_to_upload]) => {
            return get_tier_info(tier, nodes_aggregate_pool, { mirror_storage: available_to_upload[String(tier._id)] });
        });
```
but lint doesn't like it

Reply:
because the linter is asking you to get rid of the return like this:
```javascript
        .then(([nodes_aggregate_pool, available_to_upload]) =>
            get_tier_info(tier, nodes_aggregate_pool, { mirror_storage: available_to_upload[String(tier._id)] })
        );
```